lib/concurrent/edge/future.rb

```diff
@@ -486,22 +488,25 @@ def rejected?
       # @return [Object] the value of the Future when success
       def value(timeout = nil)
         touch
-        wait_until_complete timeout
-        @State.get.value
+        wait_until_complete timeout do
+          @State.get.value
+        end
```

Comment:
this should have been just `@State.get.value if wait_until_complete timeout`, which removes the expensive closure here